Rakefile

```diff
@@ -110,6 +111,31 @@ task(:commits) do
   end
 end
 
+desc "verify that changed files are clean of Ruby warnings"
+task(:warnings) do
+  # This rake task looks at all files modified in this branch. This is
+  # accomplished by using the TRAVIS_COMMIT_RANGE environment variable, which
+  # is present in travis CI and populated with the range of commits the PR
+  # contains. If not available, this falls back to `master..HEAD` as a next
+  # best bet as `master` is unlikely to ever be absent.
+  commit_range = ENV['TRAVIS_COMMIT_RANGE'].nil? ? 'master...HEAD' : ENV['TRAVIS_COMMIT_RANGE']
+  ruby_files_ok = true
+  puts "Checking modified files #{commit_range}"
+  %x{git diff --name-only #{commit_range}}.each_line do |modified_file|
+    modified_file.chomp!
+    next unless File.extname(modified_file) == '.rb'
```

Comment:
nit, I prefer `next if File.extname(...) != '.rb'` but maybe that's just me

Reply:
I don't have a strong opinion either way in this particular case. If the condition were any more complex than it is here, I'd definitely prefer the `next if ...` form, though.